src/jdk.compiler/share/classes/com/sun/tools/javac/code/Symbol.java

```diff
@@ -837,9 +837,11 @@ public final boolean precedes(TypeSymbol that, Types types) {
                 return false;
             if (type.hasTag(that.type.getTag())) {
                 if (type.hasTag(CLASS)) {
+                    int thatRank = types.rank(that.type);
+                    int thisRank = types.rank(this.type);
                     return
-                        types.rank(that.type) < types.rank(this.type) ||
-                        types.rank(that.type) == types.rank(this.type) &&
+                        thatRank < thisRank ||
+                        thatRank == thisRank &&
```

Comment:
(minor).  both before and after, the formulation is confusing and could benefit from parentheses or better indentation

Reply:
Would it be even more clear if we introduced an explicit if-else?

    int thatRank = types.rank(that.type);
    int thisRank = types.rank(this.type);
    if (thatRank < thisRank) {
        return true;
    } else if (thatRank == thisRank) {
        return that.getQualifiedName().compareTo(this.getQualifiedName()) < 0;
    }

Reply:
IMO, that's not an improvement.

Generally, style guides suggest to break before binary operators, so I'd suggest one of
```
int thatRank = types.rank(that.type);
int thisRank = types.rank(this.type);
return thatRank < thisRank
    || (thatRank == thisRank
        && that.getQualifiedName().compareTo(this.getQualifiedName()) < 0);
```

But this is now down to a more subjective opinion.   Either way is OK.

Reply:
Addressed in 140afc8.